lib/jsdom/living/nodes/HTMLFormElement-impl.js

```diff
@@ -47,14 +47,53 @@ class HTMLFormElementImpl extends HTMLElementImpl {
     super._descendantRemoved.apply(this, arguments);
   }
 
+  _getElementNodes() {
+    if (!this.hasAttributeNS(null, "id")) {
+      return domSymbolTree.treeToArray(this, {
+        filter: node => this._filterElement(node)
+      });
+    }
+
+    const id = this.getAttributeNS(null, "id");
+    return domSymbolTree.treeToArray(this.ownerDocument, {
+      filter: node => {
+        if (!this._filterElement(node)) {
+          return false;
+        }
+
+        if (this.contains(node)) {
+          return true;
+        }
+
+        if (
```

Comment:
So, the trick here is that you can replace all of this code (after the initial filter) with a call to `formOwner` from `helpers/form-controls.js`, which does the full computation. It's probably slower, but we should first go for correctness, then optimize with caches or something later if necessary.

Reply:
Thanks, that's much simpler. XD